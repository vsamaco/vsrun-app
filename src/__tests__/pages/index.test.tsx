import { describe, expect, it, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import Home from "~/pages";

describe("home page", () => {
  vi.mock("next-auth/react", () => {
    const mockSession = {
      expires: new Date(Date.now() + 2 * 86400).toISOString(),
      user: { username: "admin" },
    };
    return {
      __esModule: true,
      useSession: vi.fn(() => {
        return { data: mockSession, status: "authenticated" };
      }),
    };
  });
  it("renders page as auth", () => {
    const container = render(<Home />);
    expect(screen.getByText(/Manage Profile/)).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});

// describe.skip("home page not auth", () => {
//   it("renders page", () => {
//     vi.mock("next-auth/react", () => {
//       return {
//         __esModule: true,
//         useSession: vi.fn(() => {
//           return { data: {}, status: "unauthenticated" };
//         }),
//       };
//     });
//     render(<Home />);
//     expect(screen.getByRole("heading", { level: 1, name: "vsrun" }));
//     expect(screen.getByAltText(/Connect with Strava/)).toBeDefined();
//   });
// });
