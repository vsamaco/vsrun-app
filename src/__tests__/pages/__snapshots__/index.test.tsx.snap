// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`home page > renders page as auth 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mx-auto max-w-screen-xl px-4 py-8 text-center lg:py-16"
      >
        <h1
          class="mb-4 text-4xl font-bold leading-none tracking-tight md:mt-40 md:text-5xl lg:text-6xl"
        >
          vsrun
        </h1>
        <p
          class="mb-8 text-lg font-normal text-gray-500 dark:text-gray-400 sm:px-16 lg:px-48 lg:text-xl"
        >
          Showcase runs, stats, shoes, and races
        </p>
        <div
          class="flex flex-col items-center space-y-4 sm:flex-row sm:justify-center sm:space-x-4 sm:space-y-0"
        >
          <a
            class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
            href="/settings"
          >
            Manage Profile
          </a>
          <a
            class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 px-4 py-2"
            href="/p/milesperdonut"
          >
            Preview
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mx-auto max-w-screen-xl px-4 py-8 text-center lg:py-16"
    >
      <h1
        class="mb-4 text-4xl font-bold leading-none tracking-tight md:mt-40 md:text-5xl lg:text-6xl"
      >
        vsrun
      </h1>
      <p
        class="mb-8 text-lg font-normal text-gray-500 dark:text-gray-400 sm:px-16 lg:px-48 lg:text-xl"
      >
        Showcase runs, stats, shoes, and races
      </p>
      <div
        class="flex flex-col items-center space-y-4 sm:flex-row sm:justify-center sm:space-x-4 sm:space-y-0"
      >
        <a
          class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
          href="/settings"
        >
          Manage Profile
        </a>
        <a
          class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 px-4 py-2"
          href="/p/milesperdonut"
        >
          Preview
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
